File: 2022 czerwiec/c++/3.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <algorithm>
using namespace std;

bool is_prime(int n)
{
    if (n <= 1)
        return false;
    
    for (int i = 2; i * i <= n; i++)
        if (n % i == 0)
            return false;
    return true;
}

int main()
{
    ifstream in("liczby.txt");
    
    string s;
    while (in >> s)
    {
        int n = stoi(s);
        reverse(s.begin(), s.end());
        if (is_prime(n) && is_prime(stoi(s)))
            cout << n << '\n';
    }
}

File: 2022 czerwiec/python/3.py
with open("liczby.txt", "r") as f:
    file = f.read().strip().split('\n')

def is_prime(n):
    if n <= 1:
        return False

    for i in range(2, n):
        if n % i == 0:
            return False
    return True

for s in file:
    if is_prime(int(s)) and is_prime(int(s[::-1])):
        print(s)